src/wp-includes/class.wp-scripts.php

```diff
@@ -298,11 +292,21 @@ public function do_item( $handle, $group = false ) {
 		$after_handle  = $this->print_inline_script( $handle, 'after', false );
 
 		if ( $before_handle ) {
-			$before_handle = sprintf( "<script%s id='%s-js-before'>\n%s\n</script>\n", $this->type_attr, esc_attr( $handle ), $before_handle );
+			$before_handle = wp_inline_script(
+				$before_handle,
+				array( 'id' => esc_attr( $handle ) . '-js-before' ),
```

Comment:
See other comments. I won't comment on the other ones, `esc_attr` shouldn't be used on attributes passed to the new functions.